packages/core/src/routes/applications/application-protected-app-metadata.ts

```diff
@@ -49,7 +49,7 @@ export default function applicationProtectedAppMetadataRoutes<T extends AuthedRo
     customDomainsPathname,
     koaGuard({
       params: z.object(params),
-      status: [200, 400, 404],
+      status: [200, 400, 404, 501],
```

Comment:
should we hide 5xx and always return 500?

Reply:
Nah there are intentional 5XXs. See the assertions in this pull